internal/renderers/femtovg/lib.rs

```diff
@@ -114,21 +115,21 @@ impl FemtoVGRenderer {
     pub fn render(
         &self,
         size: PhysicalWindowSize,
-        mut before_rendering_callback: Option<impl FnOnce()>,
-    ) {
+        mut before_rendering_callback: Option<impl FnOnce() -> Result<(), PlatformError>>,
+    ) -> Result<(), i_slint_core::platform::PlatformError> {
         let width = size.width;
         let height = size.height;
 
         let canvas = if self.canvas.borrow().is_some() {
             std::cell::Ref::map(self.canvas.borrow(), |canvas_opt| canvas_opt.as_ref().unwrap())
         } else {
-            return;
+            return Ok(());
```

Comment:
looks supicious.  when would there be no canvas?

Reply:
For example if the user of the renderer API forgot to call show(). Do you think it should be an error?

Reply:
Usually, i'd say:
 - misuse of API should be panic
 - runtime error should be forwared as error.

in this case, we call render but don't render anything.  The user might wonder what's the bug, don't you think?

Reply:
I don't feel too strong about it - I think it's like calling show() on a QWidget - it'll be empty (and so will be the underlying window). Do you have a preference between (1) panic (2) runtime error (3) silence?

Reply:
I don't know. it feel like it shouldn't be silence.  Since we can report error, we might as well do.